docs/release.rst

```diff
@@ -8,6 +8,17 @@ Release notes
 
 .. _unreleased:
 
+Unreleased
+----------
+
+Fix
+~~~
+* Fix in-place mutation of input array in `BitRound`.
+  By :user:`Sam Levang <slevang>`, :issue:`608`
```

Comment:
@slevang can you please look this release note over and make any suggestions as you see fit? 🙂